stories/RefreshCache.stories.js

```diff
@@ -0,0 +1,128 @@
+    this.state = {
+      refresh: false,
+    };
+    this.refresh = this.refresh.bind(this);
+  }
+
+  refresh() {
+    this.setState(prevState => ({
+      refresh: !prevState.refresh,
+    }));
+  }
+
+  onSearchStateChange = () => {
+    this.setState({ refresh: false });
+  };
+
+  render() {
+    const displayRefresh = `${this.state.refresh}`;
+    return (
+      <InstantSearch
+        appId="latency"
+        apiKey="6be0576ff61c053d5f9a3225e2a90f76"
+        indexName="ikea"
+        refresh={this.state.refresh}
+        onSearchStateChange={this.onSearchStateChange}
+      >
+        <div>
+          <h2
+            style={{
+              color: '#182359',
+              font: 'Raleway',
+              size: '30px',
+              lineHeight: '1',
+              fontWeight: '200',
+            }}
+          >
+            Feature: Refresh cache
+          </h2>
+          <p style={{ color: '#182359', font: 'open sans' }}>
+            Adding the refresh prop to your InstantSearch component gives you
+            the possibility to refresh the cache.
+          </p>
+
+          <h3
+            style={{
+              color: '#182359',
+              font: 'Raleway',
+              size: '18px',
+              lineHeight: '1',
+              fontWeight: 'bold',
+            }}
+          >
+            How to test it?
+          </h3>
+          <div style={{ color: '#182359', font: 'open sans' }}>
+            By default, the &apos;refresh&apos; prop is disabled. You will need
+            to open your network tab in the developer tools.
+            <ol>
+              <li>
+                Type a query in the SearchBox (for instance &apos;clock&apos;).
+                You should see 5 requests to Algolia (one per letter)
+              </li>
+              <li>
+                Type &apos;clock&apos; again, you will see that no additional
+                query is made since the results are retrieved from the cache
+              </li>
+              <li>
+                Make sure the SearchBox is empty, click on the &apos;Refresh
+                cache&apos; button (you should see that Refresh set to true in
+                the info box below the SearchBox)
+              </li>
+              <li>
+                Type your previous query again: the cache has been cleared and
+                you will see new requests made to Algolia
+              </li>
+            </ol>
+          </div>
+        </div>
+        <hr />
+        <SearchBox
+          translations={{
+            placeholder: 'Search our furnitures: chairs, tables etc.',
+          }}
+        />
+        <button onClick={this.refresh} style={buttonStyle}>
+          Refresh cache
+        </button>
+        <button style={divStyle} disabled>
+          Refresh is set to: <em>{displayRefresh}</em>
+        </button>
+
+        <CustomHits />
+      </InstantSearch>
+    );
+  }
+}
+
+stories.add('with a refresh button', () => <AppWithRefresh />);
```

Comment:
Can you add the minimal jsx: 

```jsx
<InstantSearch
  appId="latency"
  apiKey="6be0576ff61c053d5f9a3225e2a90f76"
  indexName="ikea"
  refresh={this.state.refresh}
  onSearchStateChange={this.setState({ refresh: false })}
>
  <button
    onClick={() =>
      this.setState(prevState => ({ refresh: !prevState.refresh }))}
  >
    Refresh cache
  </button>
  <p>
    Refresh is set to: <em>{displayRefresh}</em>
  </p>
</InstantSearch>
```

Reply:
This story contains a `customHit` component, that's why I didn't add the JSX story (see thread on the PR to add the JSX addon)